service/service.go

```diff
@@ -115,53 +110,66 @@ func (srv *service) Shutdown(ctx context.Context) error {
 	// time should be part of configuration.
 
 	srv.telemetry.Logger.Info("Stopping receivers...")
-	if err := srv.builtReceivers.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtReceivers.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown receivers: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping processors...")
-	if err := srv.builtPipelines.ShutdownProcessors(ctx); err != nil {
+	if err := srv.host.builtPipelines.ShutdownProcessors(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown processors: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping exporters...")
-	if err := srv.builtExporters.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtExporters.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown exporters: %w", err))
 	}
 
 	srv.telemetry.Logger.Info("Stopping extensions...")
-	if err := srv.builtExtensions.ShutdownAll(ctx); err != nil {
+	if err := srv.host.builtExtensions.ShutdownAll(ctx); err != nil {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown extensions: %w", err))
 	}
 
 	return errs
 }
 
+var _ component.Host = (*serviceHost)(nil)
+
+type serviceHost struct {
+	asyncErrorChannel   chan error
+	factories           component.Factories
+	zPagesSpanProcessor *zpages.SpanProcessor
+
+	builtExporters  builder.Exporters
+	builtReceivers  builder.Receivers
+	builtPipelines  builder.BuiltPipelines
+	builtExtensions extensions.Extensions
+}
+
```

Comment:
To make the code easier to understand (the boundaries) can you move this servicehost in its own file?